Code/Framework/AzFramework/AzFramework/Quality/QualitySystemComponent.cpp

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzFramework/Quality/QualitySystemComponent.h>
+#include <AzFramework/Quality/QualityCVarGroup.h>
+
+#include <AzCore/Console/IConsole.h>
+#include <AzCore/Serialization/EditContext.h>
+#include <AzCore/Settings/SettingsRegistry.h>
+#include <AzCore/Settings/SettingsRegistryVisitorUtils.h>
+#include <AzCore/StringFunc/StringFunc.h>
+
+
+namespace AzFramework
+{
+    inline constexpr const char* QualitySettingsGroupsRootKey = "/O3DE/Quality/Groups";
+    inline constexpr const char* QualitySettingsDefaultGroupKey = "/O3DE/Quality/DefaultGroup";
+
+    // constructor and destructor defined here to prevent compiler errors
+    // if default constructor/destructor is defined in header because of
+    // the member vector of unique_ptrs
+    QualitySystemComponent::QualitySystemComponent() = default;
+    QualitySystemComponent::~QualitySystemComponent() = default;
+
+    void QualitySystemComponent::Reflect(AZ::ReflectContext* context)
+    {
+        if (AZ::SerializeContext* serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->Class<QualitySystemComponent, AZ::Component>();
+
+            if (AZ::EditContext* editContext = serializeContext->GetEditContext())
+            {
+                editContext->Class<QualitySystemComponent>(
+                    "AzFramework Quality Component", "System component responsible for quality settings")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+                    ->Attribute(AZ::Edit::Attributes::Category, "Editor")
+                    ;
+            }
+        }
+    }
+
+    void QualitySystemComponent::Activate()
+    {
+        m_settingsRegistry = AZ::SettingsRegistry::Get();
+        AZ_Assert(m_settingsRegistry, "QualitySystemComponent requires a SettingsRegistry but no instance has been created.");
+
+        m_console = AZ::Interface<AZ::IConsole>::Get();
+        AZ_Assert(m_console, "QualitySystemComponent requires an IConsole interface but no instance has been created.");

```

Comment:
Is there a need for the SettingsRegistry and Console pointers to be members of the SystemComponent?

It could be queried from the interface each time.

Holding on to a pointer doesn't make it any safer to use as it is not a reference counted object.
Actually it probably more dangerous to hold on to the pointer in this case as we have no mechanism for notification for when the Settings Registry and Console goes away.

Reply:
Good point, these pointers are left over from when I was doing a lot more logic in the system component, but I moved most of it to `QualityCVarGroup`, will remove them.